packages/salesforcedx-vscode-apex-replay-debugger/src/breakpoints/checkpointService.ts

```diff
@@ -879,6 +918,14 @@ export async function sfdxCreateCheckpoints() {
 //    and the logMessage. The logMessage is scrapped since this ends up being taken over by
 //    checkpoints for user input SOQL or Apex.
 export async function sfdxToggleCheckpoint() {
+  if (creatingCheckpoints) {
```

Comment:
I thought toggle is similar to the user creating/removing a conditional breakpoint from the gutter. Here, you disallow toggle while there's an upload in progress. But if the user does it from the gutter, you queue the action using async-lock. Why are the behaviors different?

Reply:
I actually fully described this in the description above.
"These changes are guards against the user adding/deleting/changing things while the checkpoints are being updated in the org. There are two ways this can happen, the user executes SFDX: Toggle Checkpoint while the update command is running or the user clicks in the gutter and adds/removes/changes a checkpoint. In the first case a warning message is displayed to the user since, for that, we're the ones responsible for both the breakpoint and checkpoint. In the second case, where the user manipulates one from the gutter, that was a bit tricky because we're tapping into the onDidChangeBreakpoints which means VS Code is going to do whatever and then hit the callback. For that I'd had to introduce an AsyncLock that'll prevent modifications to the checkpoint list due to breakpoint changes while a checkpoint upload was in effect but would then process the change after the checkpoint upload was finished."